File: chunk.dart
void main() {
  var arr = [1, 2, 3, 4, 5, 6, 7];
  var size = 3;
  print(chunk(arr, size));
}

chunk(arr, size) {
  var ans = [];
  var miniAns = [];
  for (var a = 0; a < arr.length; a++) {
    miniAns.add(arr[a]);
    if (miniAns.length == size) {
      ans.add(miniAns);
      miniAns = [];
    }
  }
  if (miniAns.length != 0) {
    ans.add(miniAns);
  }
  return ans;
}

File: chunk.ts
function chunks(arr: any[], size: Number): any[] {
  var ans: any[] = [];
  var miniAns: any[] = [];
  for (var a = 0; a < arr.length; a++) {
    miniAns.push(arr[a]);
    if (miniAns.length == size) {
      ans.push([...miniAns]);
      miniAns = [];
    }
  }
  if (miniAns.length != 0) {
    ans.push(miniAns);
  }
  return ans;
}

console.log(chunks([1, 2, 3, 3, 4, 5, 6, 6, 8], 3));
